### ISSUE TITLE
Update PrometheusAdapter config based on "MetricRules" objects

### ISSUE THREAD

**Issue by eero-t** [prometheus-operator/prometheus-operator#6912]
### What is missing? Please describe.

Operator updating PrometheusAdapter config based on available `MetricRules` objects, similarly to how it already updates Prometheus config based on `PodMonitor` and `ServiceMonitor` objects.

### Describe alternatives you've considered.

When one would want to automate installation of some component and its associated application metric rules...
* Helm refuses to overwrite existing config as it's already "owned" by another chart (`kube-prometheus` one), so user would need to manually edit the config
* User needs to restart PrometheusAdapther after config has been updated

This is rather aggravating and and unfriendly.  Operator providing automation for updating adapter config would IMHO make custom/external metrics much more usable!

## Environment
Kubernetes Version: 1.30
Prometheus-Operator Version: see https://github.com/prometheus-community/helm-charts/tree/main/charts/kube-prometheus-stack

**Comment by ArthurSens** [comment]
For context, Prometheus Operator automates the deployment and configuration of Prometheus and related components, so far we've been automating components that we see as essential for a Prometheus-based monitoring stack, e.g. Alertmanager.

Prometheus-Adapter can be seen as Prometheus-related component, but I wouldn't say its essential for a Prometheus-based monitoring stack. At this point in time, I'd be against adding extra complexity to this codebase to support a component that is not widely used.

Let's keep the issue open though! If we see big demand from the community, it means I'm wrong :)

**Comment by eero-t** [comment]
> Prometheus-Adapter can be seen as Prometheus-related component, but I wouldn't say its essential for a Prometheus-based monitoring stack. At this point in time, I'd be against adding extra complexity to this codebase to support a component that is not widely used.

Fair enough, but I would have thought most of the functionality to already be there:
* Monitoring creation & deletion of listed (monitor) object types
* Marshaling new monitor objects to internal structs & logging errors if that fails
* Maintaining list/map of those & creating new config when that changes
* Updating config for relevant Prometheus pod

And to need just minor changes & additions to support new object type, one containing these members: https://github.com/kubernetes-sigs/prometheus-adapter/blob/master/docs/sample-config.yaml ?

**Comment by nicolastakashi** [comment]
IIRC we already mentioned in the past about the deprecation of Prometheus Adapter, especially because it can be easily replaced by KEDA.

Maybe @slashpai can refresh me what we discussed in the past.
